src/components/AddTodo.tsx

```diff
@@ -0,0 +1,53 @@
+import { useRef } from "react";
+
+type Props = {
+  todos: {
+    id: number;
+    task: string;
+    completed: boolean;
+  }[];
+  setTodos: (
+    todo: {
+      id: number;
+      task: string;
+      completed: boolean;
+    }[]
+  ) => void;
```

Comment:
[上級FB]
（現時点で理解できなくてもいいし、修正もしなくていいです）
利用者側（このコンポーネントを利用しているApp.tsxのこと）で定義されるsetState関数をAdd/DeleteTodoで受け取っているようですね。
たしかにこれでもコードとして成立しています。

しかし、これで正しいのか考察してみます。
setTodosとは、このコンポートからみた利用者側、つまり親コンポーネントの持つsetterですが、子コンポーネントは本来親コンポートがどのような実装をされているのかを知る必要はありません。なぜならば、子コンポートとして抽象化されたコンポートは、利用者がどのような実装であるかに関わらず、あるユースケースを実現したいときに様々なシーンで利用可能であるべきだからです。

しかしながら、setTodosとしてsetterを受け取ることを指定してしまっています。これは、子コンポートであるAddTodoが利用者側の実装を指定してしまっている状態とみなす考えができます。
親コンポートは子コンポートに依存して然るべきですが、子コンポートは親コンポートに依存するべきではないということです。

ではどうするか？
子コンポートとして利用者に定義してほしいのは、「追加ボタン」をクリックしたときの挙動（つまりhandler）です。
なので本来は以下のように型定義したほうがより優れていると思います。

```suggestion
type Todo = {...}

  onClickAdd: (todos: Todo[]) => void; // またはonAddやonClickAddButtonでも
```

onHogeHogeというのは、handlerに対してよく命名で利用するjavascriptの慣習です